### PR TITLE
add new fairplay drm profile that will keep the partner's public cert…

### PR DESCRIPTION
**Pull request by noam-arad**
…ificate

### CODE REVIEW

File: plugins/drm/providers/fairplay/lib/api/KalturaFairplayProfile.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	{
+		if (is_null($dbObject))
+			$dbObject = new PlayReadyProfile();
+			
+		parent::toObject($dbObject, $skip);
+					
+		return $dbObject;
+	}
+	
+	public function validateForInsert($propertiesToSkip = array())
```
Comment (noam-arad) [inline plugins/drm/providers/fairplay/lib/api/KalturaFairplayProfile.php]: no need for validate for insert

File: plugins/drm/providers/fairplay/lib/api/KalturaFairplayProfile.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<?php
```
Comment (noam-arad) [inline plugins/drm/providers/fairplay/lib/api/KalturaFairplayProfile.php]: rename to KalturaFairplayDrmProfile

File: plugins/drm/providers/fairplay/lib/api/KalturaFairplayProfile.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	{
+		if (!PlayReadyPlugin::isAllowedPartner(kCurrentContext::getCurrentPartnerId()) || !PlayReadyPlugin::isAllowedPartner($this->partnerId))
+		{
+			throw new KalturaAPIException (KalturaErrors::PERMISSION_NOT_FOUND, 'Permission not found to use the PlayReady feature.');
+		}
+		return parent::validateForInsert($propertiesToSkip);
+	}
+	
+	public function validateForUpdate ($sourceObject, $propertiesToSkip = array())
+	{
```
Comment (noam-arad) [inline plugins/drm/providers/fairplay/lib/api/KalturaFairplayProfile.php]: no need to validate for update

File: plugins/drm/providers/fairplay/lib/model/FairplayProfile.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	// ------------------------------------------
+	// -- Custom data columns -------------------
+	// ------------------------------------------
+	
+	const CUSTOM_DATA_FAIRPLAY_PUBLIC_CERTIFICATE = 'fairplay_public_certificate';
+	
+	/**
+	 * @return string
+	 */
+	public function getPublicCertificate()
```
Comment (noam-arad) [inline plugins/drm/providers/fairplay/lib/model/FairplayProfile.php]: should be kept in file?

File: plugins/drm/providers/fairplay/admin/forms/FairplayProfileConfigureExtend.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ * @package plugins.fairplay
+ * @subpackage Admin
+ * @abstract
+ */
+class Form_FairplayProfileConfigureExtend_SubForm extends Form_DrmProfileConfigureExtend_SubForm
+{
+	public function init()
+	{
+		/*
+        $this->addElement('textarea', 'publicCertificate', array(
```
Comment (noam-arad) [inline plugins/drm/providers/fairplay/admin/forms/FairplayProfileConfigureExtend.php]: remove obsolete code

File: plugins/drm/providers/fairplay/FairplayPlugin.php [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+	/* (non-PHPdoc)
+	 * @see IKalturaPermissions::isAllowedPartner()
+	 */
+	public static function isAllowedPartner($partnerId) {
+		if (in_array($partnerId, array(Partner::ADMIN_CONSOLE_PARTNER_ID, Partner::BATCH_PARTNER_ID)))
+			return true;
+		$partner = PartnerPeer::retrieveByPK($partnerId);
+		if(!$partner)
+			return false;
+		return $partner->getPluginEnabled(self::PLUGIN_NAME);
```
Comment (noam-arad) [inline plugins/drm/providers/fairplay/FairplayPlugin.php]: need to check if we're adding fairplay configuration per partner or not

File: plugins/drm/providers/fairplay/lib/model/FairplayProfile.php (hunk 4 above)
Comment (tan-tan-kanarek) [inline plugins/drm/providers/fairplay/lib/model/FairplayProfile.php]: I see no reason, custom-data is good choice here.

File: plugins/drm/providers/fairplay/FairplayPlugin.php [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	public function contributeToEntryContextDataResult(entry $entry, KalturaEntryContextDataParams $contextDataParams, KalturaEntryContextDataResult $result)
+	{
+		if ($this->shouldContribute($entry))
+		{
+			$fairplayContextData = new KalturaFairplayEntryContextPluginData();
```
Comment (noam-arad) [inline plugins/drm/providers/fairplay/FairplayPlugin.php]: the file KalturaFairplayEntryContextPluginData is missing

File: plugins/drm/providers/fairplay/FairplayPlugin.php (hunk 6 above)
Comment (noam-arad) [inline plugins/drm/providers/fairplay/FairplayPlugin.php]: I checked, we're not

**Pull request closed without merging**